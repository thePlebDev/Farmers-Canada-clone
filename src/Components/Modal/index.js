import React,{useState} from 'react';
import styled from 'styled-components'
import CloseIcon from '@material-ui/icons/Close';
import { connect } from 'react-redux';

import SearchInput from '../SearchInput'

const CloseContainer = styled.div`

color:white;
z-index:99;
position:absolute;
right:0%;
top:0%;
transition:0.35s all;
transform:${({state})=>state ? 'translateY(0px)':'translateY(15px)'}

`

const Container = styled.div`
  visibility:${({state})=>state ? 'visible' : 'hidden'};
  width:100vw;
  height:100%;
  background-color:black;
  opacity:0;
  position:absolute;
  left:0;
  top:0;
  z-index:9;
  transition:0.35s all;
  opacity:${({state})=>state ? '0.9':'0'};

`

const Nav = styled.div`
  background-color:white;
  height:100vh;
  width:200px;
  position:absolute;
  transition:0.35s all;
  transform:${({state})=>state? 'translateX(0px)':'translateX(-200px)'};

`


const Modal = ({state,handleClick,mobileView,falseDispatcher})=>{
  console.log(mobileView)



  return(
    <Container state={mobileView}>
      <CloseContainer onClick={()=>falseDispatcher()} state={state}>
        <CloseIcon style={{fontSize:'40px'}}/>
      </CloseContainer>
      <Nav state={state}>
        <SearchInput />
      </Nav>
    </Container>
  )
}

const mapStateToProps = state => ({mobileView:state.mobileView})

const falseDispatcher = () =>{
  return({
    type:'FALSE'
  })
}


const mapDispatchToProps = {
  falseDispatcher
}

const ConnectedModal = connect(mapStateToProps,mapDispatchToProps)(Modal)
export default ConnectedModal
