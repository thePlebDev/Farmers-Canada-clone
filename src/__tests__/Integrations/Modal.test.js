import React from 'react'

import {render} from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Modal from '../../Components/Modal';


describe('it should test the modal component',()=>{
  it ('should test the snapshot',()=>{
    const {asFragment} = render(<Modal />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should test that the close button fires',()=>{
    const mockFn = jest.fn()
    let state = false

    const {getByTestId} = render(<Modal state={state} handleClick={mockFn} />)
    userEvent.click(getByTestId('close'))
    expect(mockFn.mock.calls.length).toBe(1)

  })
})
