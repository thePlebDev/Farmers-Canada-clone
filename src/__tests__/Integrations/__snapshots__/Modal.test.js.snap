// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should test the modal component should test the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw jYuqbo"
  >
    <div
      class="sc-iBPRYJ grmMKG"
      data-testid="close"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        style="font-size: 40px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </div>
    <div
      class="sc-pFZIQ khhPKQ"
    >
      <form
        class="sc-hKgILt iwQOHX"
      >
        <div
          class="sc-gsTCUz kjJXq"
        >
          <label
            for="search"
          />
          <input
            class="sc-bdfBwQ gqfphm"
            id="search"
            name="search"
            placeholder="e.g. blueberry, micro greens, Smith"
            type="text"
            value=""
          />
          <button
            class="sc-dlfnbm kghUrZ"
            data-testid="submit"
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              style="font-size: 30px; font-weight: bold;"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
        </div>
      </form>
      <div
        class="sc-jSgupP gEPNEN"
      >
        <div
          class="sc-gKsewC hRCawB"
        >
          features
        </div>
        <div
          class="sc-gKsewC hRCawB"
        >
          Network
        </div>
        <div
          class="sc-gKsewC hRCawB"
        >
          Careers
        </div>
        <div
          class="sc-gKsewC hRCawB"
        >
          Terms and Conditions
        </div>
        <div
          class="sc-gKsewC hRCawB"
        >
          Register
        </div>
        <div
          class="sc-gKsewC hRCawB"
        >
          Login
        </div>
        <div
          class="sc-eCssSg cchNuU"
        >
          Become a Vendor
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
